script/munin-httpd

```diff
@@ -35,21 +35,21 @@ package main;    ## no critic qw(Modules::ProhibitMultiplePackages)
 
 use Getopt::Long;
 
-my $port = 4948;
+my $listen = ":";
 
 print_usage_and_exit()
   unless GetOptions(
-    "port=i"  => \$port,
+    "listen=s"  => \$listen,
     "version" => \&print_version_and_exit,
     "help"    => \&print_usage_and_exit,
   );
 
-# un-taint port number
-die "ERROR: Invalid port '$port' supplied.\n" unless $port =~ /^(\d+)$/x;
-$port = $1;
+die "ERROR: Invalid listen argument supplied: '$listen'\n" unless $listen =~ /^(\S+)?:(\d+)?$/x;
+my $host = defined $1 ? $1 : "*";
+my $port = defined $2 ? $2 : 4948;
```

Comment:
Here you are using an integer (the default) or a string (parsed from the argument). Is both acceptable for `Http->new`?

Reply:
yes